lib/winapi/winbase.h

```diff
@@ -34,6 +34,35 @@ typedef VOID (WINAPI *PFLS_CALLBACK_FUNCTION)(PVOID);
 #define TLS_OUT_OF_INDEXES FLS_OUT_OF_INDEXES
 #define TLS_MINIMUM_AVAILABLE FLS_MAXIMUM_AVAILABLE
 
+typedef struct _INIT_ONCE
+{
+    PVOID Ptr;
+} INIT_ONCE, *PINIT_ONCE, *LPINIT_ONCE;
+#define INIT_ONCE_STATIC_INIT {0}
+#define INIT_ONCE_CTX_RESERVED_BITS 2
+#define INIT_ONCE_CHECK_ONLY 0x00000001UL
+#define INIT_ONCE_ASYNC 0x00000002UL
+#define INIT_ONCE_INIT_FAILED 0x00000004UL
+typedef BOOL CALLBACK PINIT_ONCE_FN(PINIT_ONCE, PVOID, PVOID *);
+
+typedef struct _CONDITION_VARIABLE {
+    INIT_ONCE initOnce;
+    PVOID eventHandles[2];
+    _Atomic int waitCount;
+} CONDITION_VARIABLE, *PCONDITION_VARIABLE;
+#define CONDITION_VARIABLE_INIT {INIT_ONCE_STATIC_INIT, {INVALID_HANDLE_VALUE, INVALID_HANDLE_VALUE}, 0}
+#define CONDITION_VARIABLE_LOCKMODE_SHARED 0x01
+
+typedef struct  _SRWLOCK
+{
+    DWORD_PTR Ptr;
+    // highest bit: reader lock (0=free, 1=locked)
+    // second-highest bit: global lock (0=free, 1=locked)
+    // remainder: blocking readers
```

Comment:
nit: I think we normally put comments about variables above the field / variable?

Reply:
Changed.